src/screens/Register/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { Box, Column, Heading, Center, FormControl, Input, Button, VStack, HStack, Link, Alert, Text } from 'native-base';
+import { useForm, Controller } from 'react-hook-form';
+import { useState } from 'react';
+import { useNavigation } from '@react-navigation/native';
+import { useRegistrationMutation, AuthUserInput, RegistrationDocument } from "../../generated/graphql";
+import { TouchableOpacity } from 'react-native';
+import * as SecureStore from 'expo-secure-store';
+
+interface FormProps {
+  email: string;
+  password: string;
+  passwordConfirmation: string;
+}
+
+function Register() {
+  const { control, handleSubmit, formState: { errors }, watch } = useForm<FormProps>();
+  const [register, { data, loading, error }] = useRegistrationMutation();
+  const [errorMessage, setErrorMessage] = useState('')
+  const navigation = useNavigation();
+
+  const emailPattern = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+  const passwordPattern = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,}$/;
+
+  const onSubmit = async (data: AuthUserInput) => {
+    const { email, password, passwordConfirmation } = data;
+
+    // Check if password fields match
+    if (password !== passwordConfirmation) {
+      Alert.alert("Passwords do not match");
+      return;
+    }
+
+    try {
+      const res = await register({
+        variables: {
+          input: {
+            email: email,
+            password: password,
+          },
+        },
+      });
```

Comment:
The way the the schemas are set up (and graphql generally goes) is that it will return a 200 if there are errors. A 200 signifies the server itself was able to process the request. In the backend im returning an object of `{ data, error }` or `{ success, error }` depending on the endpoint.
So the `catch` block is good to have, but we can also just use the `error` from `const [register, { data, loading, error }] = useRegistrationMutation();` and don't really need a setErrorMessage state hook.

In the case here, the try block will still execute if we get an error.
So instead we can base our logic off the following:
```js
error || data.error ? <error component> : null
```